src/main/java/io/yarpc/MapHandlerRegistry.java

```
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2016, Uber Technologies, Inc
+ * <p/>
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * <p/>
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ * <p/>
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package io.yarpc;
+
+
+import io.yarpc.handler.Handler;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+public class MapHandlerRegistry implements HandlerRegistry {
+
+    private final ConcurrentMap<String, Handler> handlers = new ConcurrentHashMap<String, Handler>();
+
+
+    public void addHandler(String procedure, Handler handler) {
+        handlers.put(procedure, handler);
+    }
+
+    public void removeHandler(String procedure) {
+        handlers.remove(procedure);
```

Since it's early enough in the project, I'm gonna suggest enforcing some best practices from the start. Here's one:

Use [`Objects.requireNotNull`](https://docs.oracle.com/javase/7/docs/api/java/util/Objects.html#requireNonNull(T,%20java.lang.String)) fields that can't be null when entering methods. Helps fail fast and find the exact place a null value was introduced.

Alternatively/additionally, add `javax.annotation.Nullable` and/or `javax.annotation.Nonnull` annotations where appropriate to at least help static analyzers like findbugs. (Pick one strategy and use it consistently.)